modules/dag-l1/src/main/scala/org/tessellation/dag/l1/domain/block/BlockStorage.scala

```diff
@@ -48,6 +49,18 @@ class BlockStorage[F[_]: Sync: Random](blocks: MapRef[F, ProofsHash, Option[Stor
     }.flatMap(_.liftTo[F])
       .flatMap(_ => addParentUsages(hashedBlock))
 
+  private[block] def waitingToPostpone(hashedBlock: Hashed[DAGBlock]): F[Unit] =
+    blocks(hashedBlock.proofsHash).modify {
+      case Some(_: WaitingBlock) => (PostponedBlock(hashedBlock.signed).some, hashedBlock.asRight)
+      case other                 => (other, BlockWaitingToPostponedError(hashedBlock.proofsHash, other).asLeft)
+    }.flatMap(_.liftTo[F]).void
+
+  private[block] def postponeToWaiting(proofsHash: ProofsHash, block: Signed[DAGBlock]): F[Unit] =
```

Comment:
postpone to waiting sounds like command but it's changing state from postponed to waiting 🤔

Reply:
Changed.